dbt_coves/tasks/extract/airbyte.py

```diff
@@ -0,0 +1,212 @@
+            "--host",
+            type=str,
+            required=True,
+            help="Airbyte's API hostname, i.e. 'airbyte-server'",
+        )
+        subparser.add_argument(
+            "--port",
+            type=str,
+            required=True,
+            help="Airbyte's API port, i.e. '8001'",
+        )
+        subparser.set_defaults(cls=cls, which="airbyte")
+        return subparser
+
+    def run(self):
+        self.extraction_results = {
+            "sources": set(),
+            "destinations": set(),
+            "connections": set(),
+        }
+
+        extract_destination = self.get_config_value("path")
+        airbyte_host = self.get_config_value("host")
+        airbyte_port = self.get_config_value("port")
+
+        path = pathlib.Path(extract_destination)
+
+        connections_path = path / "connections"
+        connections_path.mkdir(parents=True, exist_ok=True)
+        sources_path = path / "sources"
+        sources_path.mkdir(parents=True, exist_ok=True)
+        destinations_path = path / "destinations"
+        destinations_path.mkdir(parents=True, exist_ok=True)
+
+        self.connections_extract_destination = os.path.abspath(connections_path)
+        self.destinations_extract_destination = os.path.abspath(destinations_path)
+        self.sources_extract_destination = os.path.abspath(sources_path)
+
+        self.airbyte_api_caller = AirbyteApiCaller(airbyte_host, airbyte_port)
+
+        console.print(
+            f"Extracting Airbyte's [b]Source[/b], [b]Destination[/b] and [b]Connection[/b] configurations to {os.path.abspath(path)}\n"
+        )
+
+        dbt_sources_list = shell.run_dbt_ls(
+            "dbt ls --resource-type source",
+            None,
+        )
+        if dbt_sources_list:
+            dbt_sources_list = self._remove_airbyte_prefix(dbt_sources_list)
+            for source in dbt_sources_list:
+                # Obtain db.schema.table
+                source_db, source_schema, source_table = [
+                    element.lower() for element in source.split(".")
+                ]
+
+                source_connection = self._get_airbyte_connection_for_table(source_table)
+
+                if source_connection:
+                    source_destination = self._get_airbyte_destination_from_id(
+                        source_connection["destinationId"]
+                    )
+                    source_source = self._get_airbyte_source_from_id(
+                        source_connection["sourceId"]
+                    )
+
+                    if source_destination and source_source:
+                        self._save_json_connection(source_connection)
+                        self._save_json_destination(source_destination)
+                        self._save_json_source(source_source)
+                else:
+                    print(f"There is no Airbyte Connection for source: {source}")
+            console.print(
+                f"""Extraction successful!
```

Comment:
Could you improve this by concatenating strings so they all stay indented? i.e.
```console.print(
                "Extraction successful!"
                f"[u]Sources[/u]: {self.extraction_results['sources']}"
                ""...

Reply:
I've been reading how `rich console` works, as it seems that indentation in a multi-line string is interpreted by it. Take a look in the 'extraction results' when I indent the multi-line string correctly

![image](https://user-images.githubusercontent.com/22618882/144230821-2bfb6a5f-244c-4926-8a9a-c22d435db8e5.png)

Reply:
It doesn't seem a good practice, but using one `console.print` per line respects the console indentation
![image](https://user-images.githubusercontent.com/22618882/144231721-697e51bb-c5f9-4a99-812a-92fcdb7abcde.png)

Reply:
You´re not using my suggestion

Reply:
I followed your suggestion, but had to include `\n` at the end of each line

![image](https://user-images.githubusercontent.com/22618882/144235982-8b78fbb1-15b3-4cd9-99a8-ea4b08346495.png)

Reply:
7f7dcc2[](url)